File: mobile/lib/services/car_wash_service.dart
import '../core/api_client.dart';
import '../core/env.dart';
import '../models/booking.dart';

class CarWashService {
  final ApiClient _api = ApiClient();

  // Get all car wash bookings (staff sees only their assigned ones)
  Future<List<Booking>> getCarWashBookings() async {
    final res = await _api.getAny('${ApiEndpoints.bookings}/carwash');
    final items = <Booking>[];
    if (res is List) {
      for (final e in res) {
        if (e is Map<String, dynamic>) {
          items.add(Booking.fromJson(e));
        } else if (e is Map) {
          items.add(Booking.fromJson(Map<String, dynamic>.from(e)));
        }
      }
    }
    return items;
  }

  // Upload before wash photos
  Future<Map<String, dynamic>> uploadBeforePhotos(String bookingId, List<String> photos) async {
    return await _api.putAny(
      '${ApiEndpoints.bookings}/$bookingId/carwash/before-photos',
      body: {'photos': photos},
    );
  }

  // Upload after wash photos
  Future<Map<String, dynamic>> uploadAfterPhotos(String bookingId, List<String> photos) async {
    return await _api.putAny(
      '${ApiEndpoints.bookings}/$bookingId/carwash/after-photos',
      body: {'photos': photos},
    );
  }

  // Start car wash
  Future<Map<String, dynamic>> startCarWash(String bookingId) async {
    return await _api.putAny('${ApiEndpoints.bookings}/$bookingId/carwash/start');
  }

  // Complete car wash
  Future<Map<String, dynamic>> completeCarWash(String bookingId) async {
    return await _api.putAny('${ApiEndpoints.bookings}/$bookingId/carwash/complete');
  }
}

File: frontend/src/services/carWashService.ts
import api from './api';
import { Booking } from './bookingService';

export interface CarWashBooking extends Booking {
  carWash: {
    isCarWashService: boolean;
    beforeWashPhotos: string[];
    afterWashPhotos: string[];
    washStartedAt?: string;
    washCompletedAt?: string;
    staffAssigned?: {
      _id: string;
      name: string;
      email: string;
      phone: string;
    };
  };
}

export const carWashService = {
  // Get all car wash bookings (staff sees only their assigned ones)
  getCarWashBookings: async (): Promise<CarWashBooking[]> => {
    const response = await api.get('/bookings/carwash');
    return response.data;
  },

  // Upload before wash photos
  uploadBeforePhotos: async (bookingId: string, photos: string[]) => {
    const response = await api.put(`/bookings/${bookingId}/carwash/before-photos`, {
      photos
    });
    return response.data;
  },

  // Upload after wash photos
  uploadAfterPhotos: async (bookingId: string, photos: string[]) => {
    const response = await api.put(`/bookings/${bookingId}/carwash/after-photos`, {
      photos
    });
    return response.data;
  },

  // Start car wash
  startCarWash: async (bookingId: string) => {
    const response = await api.put(`/bookings/${bookingId}/carwash/start`);
    return response.data;
  },

  // Complete car wash
  completeCarWash: async (bookingId: string, photos?: string[]) => {
    const response = await api.put(`/bookings/${bookingId}/carwash/complete`, { photos });
    return response.data;
  },
};